### PR TITLE
[dep] upgrade serde, serde-reflection and related deps

### LINKED ISSUE #10424: aptos-core is stuck on serde < 1.0.53 due to a change in 1.0.53 that breaks testsuite/generate-format
## Repro
```
git clone git@github.com:aptos-labs/aptos-core.git
# Change serde version in Cargo.toml to 1.0.153
cd testsuite/generate-format
cargo run -- --corpus aptos
```

You will see an error like this:
```
thread 'main' panicked at 'Incomplete tracing detected inside container: StructTag:
A final registry was requested but some formats are still unknown within the container StructTag. This can
happen if `tracer.trace_value` was called on a value `foo` which does not reveal some of the underlying
formats. E.g. if a field `x` of struct `Foo` has type `Option<String>` and `foo` is value of type
`Foo` such that `foo.x == None`, then tracing the value `foo` may result in a format `Option<Unknown>`
for the field `x`. The same applies to empty vectors and empty maps.

To fix this, avoid `trace_value` and prefer `trace_type` when possible, or make sure to trace at
least one value `foo` such that `foo.x` is not empty.
', testsuite/generate-format/src/lib.rs:50:17
```

## Summary
This issue arises from this change: https://github.com/serde-rs/serde/pull/2387.

I tried using the latest version of serde + serde_yaml and it doesn't fix the issue. I also tried to fork serde and fix it but then we get a bunch of other errors, see https://github.com/banool/serde/tree/banool/undo-2387 and https://github.com/aptos-labs/aptos-core/tree/banool/upgrade-serde.

I'm not sure right now how we could handle this issue in our codebase. I believe changing the format (I'm not 100% sure what format that is, btw) is out of the question so we'd have to do something on our side to undo whatever change https://github.com/serde-rs/serde/pull/2387 introduces.

See more discussion here: https://aptos-org.slack.com/archives/C03N83P7QUC/p1691767293160939. I suspect this is relevant too: https://github.com/zefchain/serde-reflection/issues/35 (meaning switching to this version of serde-reflection might not fix our issues).

## Impact
This stops us from using crates that depend on a particular minimum version of serde higher than this version. To date this has happened twice:
- Once with me trying to use Bollard, in which case we can't because it depends on serde_with, which depends on serde 1.0.57+ (see more here: https://github.com/jonasbb/serde_with/issues/651).
- Once earlier on with @ibalajiarun in https://github.com/aptos-labs/aptos-core/pull/9463.

As time goes on this is going to happen more and more frequently as our version of serde gets increasingly out of date.

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

- This PR upgrades `serde` and `serde-reflection` crates. This change fixes #10424 and makes serde upgradable again. It uses the new API `ignore_aliases` introduced in our fork of serde-reflection https://github.com/aptos-labs/serde-reflection/pull/3 to ignore aliases in `StructTag` struct.

- This PR also upgrades `proptest` to `v1.4` and fixes test failures in aggregator tests.

- Upgrades `serde_json` and fixes related unit tests. Particularly, the latest version of `serde_json` extends the Serialize trait to include write_u128 fn, which has to be implemented to ensure that u128 are written as strings.

### Test Plan

Existing tests work after the upgrade.

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11137** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11137?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11137?utm_source=stack-comment).

**Review by gregnazario — APPROVED** [review]
When generating

File: Cargo.toml [hunk 1]
```diff
@@ -758,7 +756,6 @@ debug-assertions = true
 debug = true
 
 [patch.crates-io]
-serde-reflection = { git = "https://github.com/aptos-labs/serde-reflection", rev = "839aed62a20ddccf043c08961cfe74875741ccba" }
```
Comment (davidiw) [inline Cargo.toml:761]: I think we did this to ensure that the Move repo used this as well?

**Review by davidiw — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (ibalajiarun) [reply Cargo.toml:765]: I see. It wasn't overriding actually, so I thought it wasn't needed. Fixed it now by pointing dependency in `move-bytecode-utils` crate to workspace.

Thanks for the stamp. I need to land this first: https://github.com/aptos-labs/aptos-indexer-processors/pull/218.

**Comment by banool** [comment]
Woo!!!!!

**Pull request merged**